daemon/endpoint.go

```diff
@@ -586,9 +586,15 @@ func (d *Daemon) deleteEndpointQuiet(ep *endpoint.Endpoint, releaseIP bool) []er
 	// in its event queue.
 	ep.CloseEventQueue()
 
+	// Since the endpoint is being deleted, we need to update the caches within
+	// the policy rules.
+	// TODO - if endpoint is being regenerated here, rules cache may be updated
```

Comment:
To other reviewers - thoughts on this? Maybe I should do this within `endpointmanager.Remove` *after* we remove the endpoint from the map of endpoints which is exposed to the rest of the daemon?